hcl/hcl.go

```diff
@@ -639,28 +639,43 @@ func assignSet(name string, target *[]string, val cty.Value) error {
 		Str("action", "assignSet()").
 		Logger()
 
+	if val.IsNull() {
+		return nil
+	}
+
+	// as the parser is schemaless it only creates tuples (lists of arbitrary types).
+	// we have to check the elements themselves.
+	if !val.Type().IsTupleType() && !val.Type().IsListType() {
+		return errors.E(ErrTerramateSchema, "field %q must be a set(string) but "+
+			"found a %q", name, val.Type().FriendlyName())
+	}
+
 	logger.Trace().Msg("Iterate over values.")
 
 	errs := errors.L()
 	values := map[string]struct{}{}
 	iterator := val.ElementIterator()
+	index := -1
 	for iterator.Next() {
+		index++
 		_, elem := iterator.Element()
 
 		logger.Trace().Msg("Check element is of correct type.")
 
 		if elem.Type() != cty.String {
-			errs.Append(errors.E("field %q is a set(string) but contains %q",
-				name, elem.Type().FriendlyName()))
+			errs.Append(errors.E("field %q must be a set(string) but element %d "+
```

Comment:
```suggestion
			errs.Append(errors.E("field %q must be a string but element %d "+
```

Reply:
this generates the error: `field "wants" must be a set(string) but element 1 has type "number"`.
I'm using the terraform type declaration here.. not sure if ok, but if then it says the `wants` must be a `set` (unique values) of `string` values.